src/cryptonote_core/service_node_rules.cpp

```diff
@@ -14,15 +15,73 @@ uint64_t get_staking_requirement(cryptonote::network_type m_nettype, uint64_t he
   if (m_nettype == cryptonote::TESTNET || m_nettype == cryptonote::FAKECHAIN)
       return COIN * 100;
 
+  if (hf_version >= cryptonote::network_version_13_enforce_checkpoints)
+  {
+    constexpr uint64_t y[] = {
+        385824,
+        429024,
+        472224,
+        515424,
+        558624,
+        601824,
+        645024,
+        688224,
+        731424,
+        774624,
+        817824,
+        861024,
+        1000000,
+    };
+
+    constexpr uint64_t x[] = {
+        20458380815527,
+        19332319724305,
+        18438564443912,
+        17729190407764,
+        17166159862153,
+        16719282221956,
+        16364595203882,
+        16083079931076,
+        15859641110978,
+        15682297601941,
+        15541539965538,
+        15429820555489,
+        15000000000000,
+    };
+
+    assert(height >= y[0]);
+    constexpr uint64_t LAST_HEIGHT      = y[loki::array_count(y) - 1];
+    constexpr uint64_t LAST_REQUIREMENT = x[loki::array_count(x) - 1];
+    if (height >= LAST_HEIGHT)
+        return LAST_REQUIREMENT;
+
+    size_t i = 0;
+    for (size_t index = 0; index < loki::array_count(y); index++)
+    {
+      if (y[index] > height)
+      {
+        i = (index - 1);
+        break;
+      }
+    }
+
+    uint64_t H      = height;
+    uint64_t result = y[i] + (H - x[i]) * ((y[i + 1] - y[i]) / (x[i + 1] - x[i]));
```

Comment:
One of these differences in this expression is going to be negative (everything after the `*` is just the lsr slope, which is negative) so I'd suggesting either doing everything here in terms of `int64_t` values (which would be my preference) or else swapping the lsr difference term`(y[i+1] - y[i])` term to `(y[i] - y[i+1])` and change the first `+` to `-` in the overall expression.

Reply:
Fixed